include/rocksdb/options.h

```diff
@@ -1157,6 +1167,11 @@ struct DBOptions {
   //
   // Default: 1000000 (microseconds).
   uint64_t bgerror_resume_retry_interval = 1000000;
+
+  // Use this if your DB want to enable checksum handoff for specific file
+  // types writes. Make sure that the Env or File_system you use support the
+  // checksum verification.
+  std::vector<ChecksumHandoffFileType> checksum_handoff_file_types;
```

Comment:
This option doesn't make sense to me as written. Like Mark pointed out elsewhere, this makes it easy for the user not to get handoff checksums as expected because there are a minimum of two things to turn on to get it. Why?

At first I thought this didn't make sense at all as a DB option, but it seems plausible that DBs would have different integrity requirements for different file types. For example, some applications might only rely on WAL for fast recovery, but have an alternate method of recovery in case of corrupt WAL.

But shouldn't we have a sensible set of defaults and let Env decide whether to enable checksum handoff? If a DB option is always needed to enable it, that seems like a bad (distant, unnecessary) and dangerous (not getting what you think you're getting) abstraction.

Reply:
@pdillinger My consideration is, FileSystem gives the capability to support checksum handoff. So I plan to introduce a API to FileSystem to return the `ChecksumHandoffFuncName` (or similar). If FileSystem does not support it, or does not want to enable checksum handoff in its layer, `ChecksumHandoffFuncName` will return an empty string. Also, if the checksum function is different from the one in the RocksDB, we do not need to even calculate the checksum and send it. So, that part is like the mechanism. 

On the other side, I think we should give user/applications the option to decide, even FileSystem support it, but user/application do not want to use it, or partial use it. So I introduce the `checksum_handoff_file_types` to achieve the control. Otherwise, how can we deliver such flexibility?